modules/customize-certificates-add-service-serving-apiservice.adoc

```diff
@@ -0,0 +1,71 @@
+// Module included in the following assemblies:
+//
+// * authentication/certificates/service-serving-certificate.adoc
+
+[id="add-service-certificate-apiservice_{context}"]
+= Add the service CA bundle to an APIService
+
+////
+
+// TODO: intro + important note?
+
+A Pod can access the service CA certificate by mounting a ConfigMap that
+is annotated with `service.beta.openshift.io/inject-cabundle=true`.
+Once annotated, the cluster automatically injects the service CA
+certificate into the `service-ca.crt` key on the ConfigMap. Access to
+this CA certificate allows TLS clients to verify connections to
+services using service serving certificates.
+
+[IMPORTANT]
+====
+After adding this annotation to a ConfigMap all existing data in it is
+deleted. It is recommended to use a separate ConfigMap to contain the
+`service-ca.crt`, instead of using the same ConfigMap that stores your
+Pod's configuration.
+====
+////
+
+.Procedure
+
+. Annotate the APIService with `service.beta.openshift.io/inject-cabundle=true`.
++
+----
+$ oc annotate apiservice <apiservice-name> \//<1>
+     service.beta.openshift.io/inject-cabundle=true
+----
+<1> Replace `<apiservice-name>` with the name of the APIService to annotate.
++
+////
+TODO: Need this note for APIService?
+[NOTE]
+====
+Explicitly referencing the `service-ca.crt` key in a volumeMount will
+prevent a Pod from starting until the ConfigMap has been injected with
+the CA bundle.
+====
+////
++
+For instance, to annotate the APIService `foo` the following command would be
+used:
++
+----
+$ oc annotate apiservice foo service.beta.openshift.io/inject-cabundle=true
+----
+
+. View the APIService to ensure the certificate has been generated. This
+appears as `caBundle` in the YAML output.
++
+----
+$ oc get apiservice <apiservice-name> -o yaml
+apiVersion: apiregistration.k8s.io/v1
+kind: APIService
+metadata:
+  annotations:
+    service.beta.openshift.io/inject-cabundle: "true"
+...
+spec:
+  caBundle: <PEM_ENCODED_CA_BUNDLE>
+...
+----
+
+// TODO: Confirm it's PEM encoded ^
```

Comment:
Yes, PEM encoded. The `-----BEGIN CERTIFICATE-----` header is the giveaway.

Reply:
@marun The output I see doesn't have the `-----BEGIN CERTIFICATE-----` for all of these `caBundle` fields.

Reply:
@marun What I see is what's listed on this page https://github.com/openshift/service-ca-operator/ under the "generic cabundle injector": `caBundle: LS0tLS1CRUdJTiBDRVJUS...`. I see this for all except ConfigMap, which does show "BEGIN CERTIFICATE".

Can you confirm that that's correct, and if so, whether the `caBundle` fields are still PEM-encoded, or should I just say base64 encoded?